src/js/legend_control.js

```diff
@@ -0,0 +1,593 @@
+
+    return row;
+  }
+
+  /**
+   * Returns an HTML image element displaying the given style image.
+   *
+   * @param styleImage The style image to display.
+   * @param iconSize The size factor to apply to the width and height of the image.
+   * @returns An HTML image element, or nothing if the style image is merely a spacer image.
+   */
+  getImageFromStyle(styleImage, iconSize = 1) {
+    let userImage = styleImage.userImage;
+    // Skip spacer images representing unsupported networks.
+    if (userImage?.width === 1 || userImage?.height === 1) return;
+
+    let imageData = new ImageData(
+      userImage?.data || new Uint8ClampedArray(styleImage.data.data),
+      userImage?.width || styleImage.data.width,
+      userImage?.height || styleImage.data.height
+    );
+
+    let canvas = document.createElement("canvas");
+    canvas.width = imageData.width;
+    canvas.height = imageData.height;
+    let ctx = canvas.getContext("2d");
+    ctx.putImageData(imageData, 0, 0);
+    let img = new Image(
+      (imageData.width * iconSize) / ShieldDraw.PXR,
+      (imageData.height * iconSize) / ShieldDraw.PXR
+    );
+    img.src = canvas.toDataURL("image/png");
+    img.className = "shield";
+
+    return img;
+  }
+
+  /**
+   * Inserts human-readable descriptions in each of the given table rows.
+   *
+   * @param rows An array of table rows containing placeholders for descriptions.
+   */
+  async prettifyNetworkLabels(rows) {
+    let networkMetadata = await this.getNetworkMetadata();
+    if (!networkMetadata) return;
+
+    let toTitleCase = (s) => s[0].toLocaleUpperCase() + s.substring(1);
+    for (let row of rows) {
+      let network = row.dataset.network;
+      let binding = networkMetadata[network];
+      if (!binding) continue;
+
+      let descriptionCell = row.querySelector(".description");
+      descriptionCell.setAttribute("lang", binding.networkLabel["xml:lang"]);
+      descriptionCell.textContent = toTitleCase(binding.networkLabel.value);
+    }
+  }
+
+  /**
+   * Returns a mapping from `network=*` values to metadata about these values from Wikidata.
+   */
+  async getNetworkMetadata() {
+    if (this._networkMetadata) {
+      return this._networkMetadata;
+    }
+
+    let url = `https://query.wikidata.org/sparql?query=${encodeURIComponent(
+      this.getNetworkMetadataQuery()
+    )}&format=json`;
+    const response = await fetch(url);
+    const json = await response.json();
+    this._networkMetadata = Object.fromEntries(
+      json.results.bindings.map((binding) => {
+        return [binding.value.value, binding];
+      })
+    );
+    return this._networkMetadata;
+  }
+
+  purgeNetworkMetadata() {
+    delete this._networkMetadata;
+  }
+
+  /**
+   * Returns the Wikidata Query Service SPARQL query for network metadata.
+   */
+  getNetworkMetadataQuery() {
+    let locales = Label.getLocales().join(",");
+    return `
+SELECT ?value ?network ?networkLabel WHERE {
+  ?network wdt:P1282 ?tag.
```

Comment:
I considered limiting this query to instances of [highway system (Q25631158)](https://www.wikidata.org/wiki/Q25631158) or [road (Q34442)](https://www.wikidata.org/wiki/Q34442), to avoid confusion with list articles about these networks and also filter out all the `network=*` values about railway and public transportation networks. However, I suspect that would make the query take longer and it’d be wasteful. Maybe once we move this query to a build step, we can make it more sophisticated and targeted.